src/main/src/test/java/org/geoserver/GeoServerLifecycleTests.java

```diff
@@ -0,0 +1,27 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class GeoServerLifecycleTests extends GeoServerLifecycleTest {
```

Comment:
Confusing and wasteful:
- Since it extends another class, the tests in the base class are run again (so testRunning is really run twice)
- Having two classes named GeoServerLifecycleTest and GeoServerLifcycleTests is also confusing, does not show what is the reason for the two classes, the explanation provided in the PR will be lost, git history does not retain it.

It seems that what you need is a clean restart of the full class at every method run, which you can get here:

- https://github.com/geoserver/geoserver/blob/main/src/main/src/test/java/org/geoserver/test/GeoServerSystemTestSupport.java#L172
I

If that does not solve the issue, the explanation needs to go in the test classes javadoc and better names should be chosen to explain the reason. GeoServer is a long lived project, code must be written so that it's somewhat understandable years down the line, by someone else that would not have been around when this PR was reviewed.